src/Model/TopEdits.php

```diff
@@ -155,6 +155,34 @@ public function getNumTopEdits(): int
         return count($this->topEdits);
     }
 
+    /**
+     * Get the WikiProject totals.
+     * @param int Namespace ID.
+     * @return string[]|int
+     */
+    public function getProjectTotals(int $ns) : array
+    {
+        $projectTotals = [];
+        // List of pages for this namespace
+        $rows = $this->topEdits[$ns];
+        foreach ($rows as $row) {
+            $num = $row["count"];
+            // May be null or nonexistent for assessment-less pages
+            $titles = $row["pap_project_title"] ?? "{}";
+            // Had to use json to pass multiple values in SQL select
+            foreach (JSON_decode($titles) as $projectName) {
```

Comment:
```suggestion
            foreach (json_decode($titles) as $projectName) {
```
that's the [documented](https://www.php.net/manual/en/function.json-decode.php) form. Maybe we should add a linter rule for this.

Reply:
Done